flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java

```diff
@@ -288,7 +288,7 @@ public MySqlSourceConfig createConfig(int subtaskId) {
         props.put("bigint.unsigned.handling.mode", "precise");
 
         if (serverIdRange != null) {
-            int serverId = serverIdRange.getServerId(subtaskId);
+            long serverId = serverIdRange.getServerId(subtaskId);
```

Comment:
Server id has a fixed range in mysql (https://dev.mysql.com/doc/refman/5.7/en/replication-options.html).
I think we should check the config option of the server id range at the method `ServerIdRange#from`.